components/_impl/workers/subprocess_worker.py

```diff
@@ -117,6 +117,11 @@ def __init__(self, timeout: typing.Optional[float] = None) -> None:
             **popen_kwargs,
         )
 
+        # setup the pid of child process for the pipes
```

Comment:
The proc's pid is unknown when we build the pipe, so reset the correct pids here.